### PR TITLE
refactor logic related to products

### PR DESCRIPTION
**Pull request by yx2512**

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -294,288 +318,192 @@ def post(self):
...
+class ProductResource(Resource):
   """
-  Delete a product from a wishlist
-  This endpoint will delete a product from a wishlist based the wishlist_id
-  and the product_id specified in the URL
+  ProductResource class
+
+  Allows the manipulation on a single Product in a Wishlist
+  GET - Returns a product in a wishlist
+  POST - Create a product in a wishlist
```
Comment (ntlz) [inline service/routes.py:412]: there is no post in this class.

**Review by ntlz — CHANGES_REQUESTED** [review]

File: service/routes.py [hunk 2]
```diff
@@ -294,288 +318,192 @@ def post(self):
...
-  POST - Create the product in the wishlist
-  DELETE - Delete the product from the wishlist
+  Allows the manipulation on a collection of Products in a Wishlist
+  POST - Create a product in a wishlist
   """
+  @api.doc('create_a_product')
+  @api.response(415,"Unsupported media type : application/json expected")
+  @api.response(400,"Expected a json request body")
+  @api.expect(create_product_model)
+  def post(self,wishlist_id):
```
Comment (XinyuC97) [inline service/routes.py:350]: I think this function will return a code ```201``` and the created product.

File: service/routes.py [hunk 3]
```diff
@@ -113,6 +114,28 @@ def info():
...
+    enum=Availability._member_names_),
+  'pic_url': fields.String(required=False,
+    description='URL for a picture of the product.'),
+  'short_desc': fields.String(required=False,
+    description='Short description of the product.'),
+  'inventory_product_id': fields.Integer(required=False,
+    description='ID of the product in inventory.'),
+  'wishlist_id': fields.Integer(required=False,
+    description='Wishlist ID that this product belongs to.'),
+  'in_cart_status': fields.String(required=False,
```
[the remark this answers is not in the document]
Comment (yx2512) [reply service/routes.py]: solved

File: service/routes.py [hunk 4]
```diff
@@ -294,288 +318,192 @@ def post(self):
...
-  GET / - Returns all products in a wishlists in the database
-  DELETE / - Delete all products in a wishlist
-  """
+    product = Product()
+    product.deserialize(data)
+    product.create()
 
-  def get(self):
-    pass
+    return product.id, status.HTTP_201_CREATED
```
[the remark this answers is not in the document]
Comment (yx2512) [reply service/routes.py]: solved

File: service/routes.py (hunk 1 above)
Comment (yx2512) [reply service/routes.py]: solved

File: service/routes.py (hunk 2 above)
Comment (yx2512) [reply service/routes.py]: solved

**Review by XinyuC97 — APPROVED** [review]

**Pull request merged**